File: utility-billing-system/src/app/core/models/payment.ts
// payment.models.ts - Add to core/models/

export interface MakePaymentRequest {
  billId: number;
  amount: number;
  paymentMode: number; // 1=Cash, 2=Online, 3=Cheque
  paymentMethod: string; // "UPI", "Card", "NetBanking", etc.
}

export interface PaymentResponse {
  success: boolean;
  message: string;
  transactionId: string;
  receiptNumber: string;
  payment: Payment;
}

export interface Payment {
  paymentId: number;
  receiptNumber: string;
  consumerId: number;
  consumerName: string;
  billId: number;
  billNumber: string;
  amount: number;
  paymentMode: string; // "Online", "Cash", "Cheque"
  transactionId: string | null;
  paymentDate: string;
  remarks: string;
}

export enum PaymentMode {
  Cash = 1,
  Online = 2,
  Cheque = 3
}

export enum PaymentMethod {
  UPI = 'UPI',
  Card = 'Card',
  NetBanking = 'NetBanking',
  Cash = 'Cash',
  Cheque = 'Cheque'
}

File: Backend/UtilityBillingSystem.Core/Entities/Payment.cs
using UtilityBillingSystem.Core.Enums;

namespace UtilityBillingSystem.Core.Entities
{
    public class Payment
    {
        public int PaymentId { get; set; }
        public string ReceiptNumber { get; set; }
        public int ConsumerId { get; set; }
        public int BillId { get; set; }
        public decimal Amount { get; set; }
        public PaymentMode PaymentMode { get; set; }
        public string? TransactionId { get; set; }
        public DateTime PaymentDate { get; set; }
        public string Remarks { get; set; }
        public int RecordedBy { get; set; }
        public DateTime CreatedAt { get; set; }

        // Navigation
        public Consumer Consumer { get; set; }
        public Bill Bill { get; set; }
        public User RecordedByUser { get; set; }
    }
}